### PR TITLE
Add Avatto ZWSH16 Smart Temperature and Humidity Detector device defi…

### PR DESCRIPTION
**Pull request by shadowpanther**
…nition

### CODE REVIEW

**Comment by shadowpanther** [comment]
Bought one from [here](https://www.aliexpress.com/item/1005007288600628.html)
Got it to work by guessing the Tuya datapoints.

File: src/devices/avatto.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+            const endpoint = device.getEndpoint(1);
+            await tuya.configureMagicPacket(device, coordinatorEndpoint);
+            await endpoint.command('manuSpecificTuya', 'mcuVersionRequest', {seq: 0x0002});
+        },
+        exposes: [e.battery(), e.temperature(), e.humidity()],
+        meta: {
+            tuyaDatapoints: [
+                [1, 'temperature', tuya.valueConverter.divideBy10],
+                [2, 'humidity', tuya.valueConverter.raw],
+                [9, 'temperature_scale', tuya.valueConverterBasic.lookup({celsius: 0, fahrenheit: 1})],
```
Comment (Koenkk) [inline src/devices/avatto.ts:26]:
```suggestion
                [9, 'temperature_unit', tuya.valueConverter.temperatureUnit],
```

File: src/devices/avatto.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        description: 'Smart Temperature and Humidity Detector',
+        fromZigbee: [tuya.fz.datapoints],
+        toZigbee: [tuya.tz.datapoints],
+        onEvent: tuya.onEventSetTime,
+        configure: async (device, coordinatorEndpoint) => {
+            const endpoint = device.getEndpoint(1);
+            await tuya.configureMagicPacket(device, coordinatorEndpoint);
+            await endpoint.command('manuSpecificTuya', 'mcuVersionRequest', {seq: 0x0002});
+        },
+        exposes: [e.battery(), e.temperature(), e.humidity()],
```
Comment (Koenkk) [inline src/devices/avatto.ts:21]:
```suggestion
        exposes: [e.battery(), e.temperature(), e.humidity(), tuya.exposes.temperatureUnit(),  tuya.exposes.batteryState()],
```

**Comment by shadowpanther** [comment]
Added your suggestions, thank you!

**Comment by Koenkk** [comment]
thanks!

**Comment by sch115** [comment]
@shadowpanther Hi! Just going through major issues with battery stated as null (not reporting). Did some searches via Tuya gateway and found out that **battery level** datapoint is 4 and **battery state** is unused. 14 and 15 stated in your code is for Temp alarm and Humidity alarm. My knowledge is far from perfect in this case, so I have no idea how to test it. Would you please be helpful in this case? I would really appreciate that! Thx!

https://github.com/Koenkk/zigbee2mqtt/issues/30750

**Pull request merged**